sh/remove_signer.sh

```diff
@@ -0,0 +1,194 @@
+    fi
+}
+
+_assert_no_path_cycles() {
+    local target path
+
+    if [ $# -gt 16 ]; then
+        return 1
+    fi
+
+    target="$1"
+    shift
+
+    for path in "$@"; do
+        if [ "$path" = "$target" ]; then
+            return 1
+        fi
+    done
+}
+
+_canonicalize() {
+    local d f
+    if [ -d "$1" ]; then
+        (CDPATH= cd -P "$1" 2>/dev/null && pwd -P)
+    else
+        d="$(_directory "$1")"
+        f="$(_file "$1")"
+        (CDPATH= cd -P "$d" 2>/dev/null && printf '%s/%s\n' "$(pwd -P)" "$f")
+    fi
+}
+
+## end POSIX Bash implementation of realpath
+
+set -Eeufo pipefail -o posix
+
+declare project_root
+project_root="$(_directory "$(_directory "$(realpath "${BASH_SOURCE[0]}")")")"
+declare -r project_root
+cd "$project_root"
+
+. "$project_root"/sh/common.sh
+
+declare safe_address
+safe_address="$(get_config governance.deploymentSafe)"
+declare -r safe_address
+
+. "$project_root"/sh/common_safe.sh
+
+declare signer
+IFS='' read -p 'What address will you submit with?: ' -e -r -i 0xEf37aD2BACD70119F141140f7B5E46Cd53a65fc4 signer
+declare -r signer
+
+. "$project_root"/sh/common_safe_deployer.sh
+. "$project_root"/sh/common_wallet_type.sh
+
+declare old_owner
+old_owner="$1"
+shift
+old_owner="$(cast to-checksum "$old_owner")"
+declare -r old_owner
+
+declare -r getThreshold_sig='getThreshold()(uint256)'
+declare -i threshold
+threshold="$(cast call --rpc-url "$rpc_url" "$safe_address" "$getThreshold_sig")"
+declare -r -i threshold
+
+declare -r removeOwner_sig='removeOwner(address,address,uint256)'
+declare removeOwner_call
+removeOwner_call="$(cast calldata removeOwner_sig "$(prev_owner "$old_owner")" "$old_owner" "$threshold")"
+declare -r removeOwner_call
+
+# set minimum gas price to (mostly for Arbitrum and BNB)
+declare -i min_gas_price
+min_gas_price="$(get_config minGasPriceGwei)"
+min_gas_price=$((min_gas_price * 1000000000))
+declare -r -i min_gas_price
+declare -i gas_price
+gas_price="$(cast gas-price --rpc-url "$rpc_url")"
+if (( gas_price < min_gas_price )) ; then
+    echo 'Setting gas price to minimum of '$((min_gas_price / 1000000000))' gwei' >&2
+    gas_price=$min_gas_price
+fi
+declare -r -i gas_price
+declare -i gas_estimate_multiplier
+gas_estimate_multiplier="$(get_config gasMultiplierPercent)"
+declare -r -i gas_estimate_multiplier
+
+declare signing_hash
+signing_hash="$(eip712_hash "$removeOwner_call" 0 "$safe_address")"
+declare -r signing_hash
+
+declare packed_signatures
+packed_signatures="$(retrieve_signatures remove_signer "$removeOwner_call" 0 "$safe_address")"
+declare -r packed_signatures
+
+# configure gas limit
+declare -r -a args=(
+    "$safe_address" "$execTransaction_sig"
+    # to, value, data, operation, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver, signatures
+    "$safe_address" 0 "$removeOwner_call" 0 0 0 0 "$(cast address-zero)" "$(cast address-zero)" "$packed_signatures"
+)
+
+# set gas limit and add multiplier/headroom (again mostly for Arbitrum)
```

Comment:
probably need this for Mantle too

Reply:
We merely double the gas limit on Mantle and set the gas price to 1 gwei. What do you think are reasonable values there?

Reply:
Mantle is OP-stack so shouldn't the DA fee be billed directly in MNT rather than gas? And we do use `eth_estimateGas` to figure out the gas consumption, so the buffer is only there to compensate for errors in that RPC. Do we really need to 3kx the gas limit (not just the price) in order to get reasonable execution?

Unless you mean that we should increase the gas price substantially?

Reply:
oh i see. if we are calling `eth_estimateGas` then it should be fine! 

Mantle deviated a lot from OP-stack and they charge DA fee explicitly (similar to how Arbitrum does).